Automattic/wp-calypso
# Hide masterbar when Calypso is viewed from WC apps

#### Changes proposed in this Pull Request

* Add a new utility method, `isWcMobileApp`, to detect when Calypso is being displayed from within the WooCommerce apps (WCiOS or WCAndroid).
* Hide the `masterbar` if Calypso is accessed from the WooCommerce app (WPiOS or WCAndroid).

| Before | After |
| - | - |
| <img width="250" alt="Before: the masterbar is shown" src="https://user-images.githubusercontent.com/1898325/124011602-31ebb580-d9ae-11eb-9e68-6a2ef0ae21cd.png"> | <img width="250" alt="After: the masterbar is hidden" src="https://user-images.githubusercontent.com/1898325/124011554-23050300-d9ae-11eb-9fe7-dc4c4d6c3412.png"> |


#### Testing instructions

* Checkout this branch and run the app locally: `yarn && yarn start`
* Open http://calypso.localhost:3000
* Verify the masterbar is visible as usual
* Change the [user agent to that of WCiOS](https://github.com/woocommerce/woocommerce-ios/blob/4fe952211777fad134084cbc19d4db229a37c8c9/Networking/Networking/Settings/UserAgent.swift#L40): `Mozilla/5.0 (iPhone; CPU iPhone OS 12_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/16B91 wc-ios/12.1`
* Reload the page and verify that the masterbar is hidden
* Change the user agent to that of WCAndroid: `Mozilla/5.0 (iPhone; CPU iPhone OS 12_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/16B91 wc-android/12.1`
* Reload the page and verify that the masterbar is still hidden

Tip: To change the user agent in Chrome, see below screenshots (taken from https://github.com/Automattic/wp-calypso/pull/31830):

<img width="404" alt="55129228-ab562700-5159-11e9-9a75-d05cfe97b65e" src="https://user-images.githubusercontent.com/1898325/124031455-556e2a80-d9c5-11eb-9a74-02a5315f4bf9.png">
<img width="900" alt="55129288-df314c80-5159-11e9-8278-f14471c9eb2f" src="https://user-images.githubusercontent.com/1898325/124031451-54d59400-d9c5-11eb-9924-d252d408879f.png">

Fixes #53982
